src/views/FlowView.tsx

```diff
@@ -148,174 +122,128 @@ class FlowView extends React.Component<any, any> {
     });
 
     newChart.nodes[node.id].properties = props;
+    dispatch(updateFlow({ ...chart, ...newChart }));
 
-    this.updateFlow({ ...chart, ...newChart }, "updateNode");
     return newChart.nodes[node.id];
   };
 
-  updateLink = (linkId: string, fromId: string, toId: string | undefined) => {
+  const updateLink = (
+    linkId: string,
+    fromId: string,
+    toId: string | undefined
+  ) => {
     if (fromId === toId) return;
-    let { chart } = this.state;
     let newChart = cloneDeep(chart);
 
     newChart.links[linkId].from.nodeId = fromId;
     newChart.links[linkId].to.nodeId = toId;
 
-    this.updateFlow({ ...chart, ...newChart }, "updateLink");
-  };
-
-  deleteSelection = () => {
-    this.state.actionCallbacks.onDeleteKey({});
-  };
-
-  updateFlow = (flow: any, event: any) => {
-    if (syncEvents.includes(event)) return this.syncFlow(flow);
-    this.setState({ chart: flow });
+    dispatch(updateFlow({ ...chart, ...newChart }));
   };
 
-  syncFlow = (flow: Flow) => {
-    this.props.updateFlow(flow);
-  };
-  //todo can't find usages. Maybe we can delete that
-  selectNode = (data: any) => {
-    Dispatcher.dispatch({
-      actionType: Constants.SELECT_NODE,
-      payload: data.nodeId,
-    });
+  const deleteSelection = () => {
+    actionCallbacks.onDeleteKey({});
   };
 
-  copyChartAsYAML = () => {
-    copyToClipboard(formatAsYAML(this.props.chart));
+  const copyChartAsYAML = () => {
+    copyToClipboard(formatAsYAML(chart));
     alert("Chart copied to clipboard as YAML");
   };
-
-  validateLink = ({ fromNodeId, toNodeId, fromPortId, toPortId }: any) => {
+  const validateLink = ({
+    fromNodeId,
+    toNodeId,
+    fromPortId,
+    toPortId,
+  }: any) => {
```

Comment:
Do we have to go with `any` here?

Reply:
I copied it from the original implementation. Changed the type to IOnLinkCompleteInput from the package